contracts/swap/swap.go

```diff
@@ -169,11 +177,35 @@ func (s simpleContract) LiquidBalance(opts *bind.CallOpts) (*big.Int, error) {
 	return s.instance.LiquidBalance(opts)
 }
 
+//Token returns the address of the ERC20 contract, used by the chequebook
+func (s simpleContract) Token(opts *bind.CallOpts) (common.Address, error) {
+	return s.instance.Token(opts)
+}
+
+//BalanceOf returns the balance of the account for the underlying ERC20 contract of the chequebook
+func (s simpleContract) BalanceOf(opts *bind.CallOpts, account common.Address) (*big.Int, error) {
```

Comment:
I would either rename this or put it on a different receiver as this is not actually a chequebook related call.

Reply:
`func (s simpleContract) BalanceAtTokenContract(opts *bind.CallOpts, account common.Address) (*big.Int, error)`